scripts/tests/chiptest/yamltest_with_chip_repl_tester.py

```diff
@@ -0,0 +1,114 @@
+#
+#    Copyright (c) 2022 Project CHIP Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import click
+import functools
+import glob
+import os
+import tempfile
+
+from chip import ChipDeviceCtrl
+from chip.ChipStack import *
+import chip.FabricAdmin
+import chip.CertificateAuthority
+import chip.native
+from chip.yaml.runner import ReplTestRunner
+
+from matter_yamltests.definitions import ParseSource, SpecDefinitions
+from matter_yamltests.parser import TestParser
+
+
+_DEFAULT_CHIP_ROOT = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), "..", "..", ".."))
+_CLUSTER_XML_DIRECOTRY_PATH = os.path.abspath(
+    os.path.join(_DEFAULT_CHIP_ROOT, "src/app/zap-templates/zcl/data-model/"))
+
+
+def _sort_with_global_attribute_first(a, b):
+    if a.endswith('global-attributes.xml'):
+        return -1
+    elif b.endswith('global-attributes.xml'):
+        return 1
+    elif a > b:
+        return 1
+    elif a == b:
+        return 0
+    elif a < b:
+        return -1
+
+
+@click.command()
+@click.option(
+    '--setup-code',
+    default=None,
+    help='setup-code')
+@click.option(
+    '--yaml-path',
+    default=None,
+    help='yaml-path')
+@click.option(
+    '--node-id',
+    default=0x12344321,
+    help='Node ID to use when commissioning device')
+def main(setup_code, yaml_path, node_id):
+    # Setting up python enviroment for running YAML CI tests using python parser.
+    chip_stack_storage = tempfile.NamedTemporaryFile()
```

Comment:
should we use this as a context via `with tempfile.NamedTemporaryFile() as chip_stack_storage` to ensure cleanup after execution?